### PR TITLE
ADAL and MSAL coexistence

### PR DESCRIPTION
**Pull request by antrix1989**

### CODE REVIEW

File: ADAL/src/ADAuthenticationContext.m [hunk 1]
```diff
@@ -234,7 +236,17 @@ + (BOOL)isResponseFromBroker:(NSString *)sourceApplication
     isBroker = isBroker || [sourceApplication isEqualToString:ADAL_BROKER_APP_BUNDLE_ID_DOGFOOD];
 #endif
     
-    return response && isBroker;
+    NSURLComponents *components = [NSURLComponents componentsWithURL:response resolvingAgainstBaseURL:NO];
+    NSString *qp = [components percentEncodedQuery];
+    NSDictionary *queryParamsMap = [NSDictionary adURLFormDecode:qp];
+    
+    NSString *protocolVersion = queryParamsMap[@"msg_protocol_ver"];
+    BOOL isValidVersion = [protocolVersion isEqualToString:@"2"];
```
Comment (oldalton) [inline ADAL/src/ADAuthenticationContext.m]: nit: can we declare it as a constant somewhere?

File: ADAL/src/ADAuthenticationContext.m [hunk 2]
```diff
@@ -234,7 +236,17 @@ + (BOOL)isResponseFromBroker:(NSString *)sourceApplication
...
+    
+    NSDictionary *resumeDictionary = [[NSUserDefaults standardUserDefaults] objectForKey:kAdalResumeDictionaryKey];
+    BOOL isADALInitiatedRequest = [resumeDictionary[kAdalSDKNameKey] isEqualToString:kAdalSDKObjc];
+    
+    return response && isBroker && isValidVersion && isADALInitiatedRequest;
```
Comment (oldalton) [inline ADAL/src/ADAuthenticationContext.m]: Nit: method name "isResponseFromBroker" indicates we're checking if it's a response from broker only. Now we're also checking if response is for us. Can we rename the method or add a separate one indicating its purpose?

File: ADAL/src/ADAuthenticationContext.m (hunk 1 above)
Comment (antrix1989) [reply ADAL/src/ADAuthenticationContext.m]: done

File: ADAL/src/ADAuthenticationContext.m (hunk 2 above)
Comment (antrix1989) [reply ADAL/src/ADAuthenticationContext.m]: Added a separate method.

**Pull request merged**